hazelcast/src/main/java/com/hazelcast/spi/discovery/integration/DiscoveryServiceProvider.java

```diff
@@ -25,18 +27,23 @@
  * provide this ability. Every service should have its own provider, however in rare cases a single provider might
  * create different <tt>DiscoveryService</tt> implementations based on the provided {@link DiscoveryMode} or other
  * configuration details.
+ *
+ * @since 3.6
  */
+@Beta
 public interface DiscoveryServiceProvider {
 
     /**
      * Instantiates a new instance of the {@link DiscoveryService}.
      *
-     * @param discoveryMode            the current discovery mode
+     * @param member                    the current <tt>Member</tt> instance if created on member nodes, otherwise <tt>null</tt>
+     * @param logger                    the <tt>ILogger</tt> instance to be used for logging output
+     * @param discoveryMode             the current discovery mode
      * @param discoveryStrategiesConfig the configuration parsed from the XML or provided using the programmatic API
-     * @param configClassLoader        the classloader set in the configuration
+     * @param configClassLoader         the classloader set in the configuration
      * @return a new instance of the discovery service
      */
-    DiscoveryService newDiscoveryService(DiscoveryMode discoveryMode, DiscoveryStrategiesConfig discoveryStrategiesConfig,
-                                         ClassLoader configClassLoader);
+    DiscoveryService newDiscoveryService(Member member, ILogger logger, DiscoveryMode discoveryMode,
+                                         DiscoveryStrategiesConfig discoveryStrategiesConfig, ClassLoader configClassLoader);
```

Comment:
I think he meant you could rename the variable `discoveryStrategiesConfig` to `config` to make it shorter, since it's clear what kind of config it is due to the context.